frontend/src/lib/components/TaxonomicFilter/taxonomicFilterLogic.test.ts

```diff
@@ -52,140 +60,193 @@ describe('taxonomicFilterLogic', () => {
         for (const listGroupType of logicProps.taxonomicGroupTypes) {
             infiniteListLogic({ ...logicProps, listGroupType }).mount()
         }
-    })
+    }
 
-    it('mounts all infinite list logics', async () => {
-        await expectLogic(logic).toMount([
-            infiniteListLogic({ ...logic.props, listGroupType: TaxonomicFilterGroupType.Events }),
-            infiniteListLogic({ ...logic.props, listGroupType: TaxonomicFilterGroupType.Actions }),
-            infiniteListLogic({ ...logic.props, listGroupType: TaxonomicFilterGroupType.Elements }),
-        ])
-        expect(
-            infiniteListLogic({ ...logic.props, listGroupType: TaxonomicFilterGroupType.Cohorts }).isMounted()
-        ).toBeFalsy()
-    })
+    describe('when set to not clear the search query on selection', () => {
+        beforeEach(() => {
+            setupLogic(false)
+        })
 
-    it('keeps infiniteListCounts in sync', async () => {
-        await expectLogic(logic)
-            .toMatchValues({
-                infiniteListCounts: {
-                    [TaxonomicFilterGroupType.Events]: 0,
-                    [TaxonomicFilterGroupType.Actions]: 0,
-                    [TaxonomicFilterGroupType.Elements]: 4,
-                },
+        it('can select an item without clearing search query', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.setSearchQuery('tomato')
+                logic.actions.selectItem({} as TaxonomicFilterGroup, '' as TaxonomicFilterValue, null)
+            }).toMatchValues({
+                searchQuery: 'tomato',
             })
-            .toDispatchActions(['infiniteListResultsReceived'])
-            .delay(1)
-            .clearHistory()
-            .toMatchValues({
-                infiniteListCounts: {
-                    [TaxonomicFilterGroupType.Events]: 56,
-                    [TaxonomicFilterGroupType.Actions]: 0, // not mocked
-                    [TaxonomicFilterGroupType.Elements]: 4,
-                },
+        })
+
+        it('clears the search query when the column configurator saves', async () => {
+            await expectLogic(logic, () => {
+                logic.actions.setSearchQuery('tomato')
+                columnConfiguratorLogic.actions.save()
+            }).toMatchValues({
+                searchQuery: '',
             })
+        })
```

Comment:
Wouldn't this fit better into a column configurator test? This taxonomic filter should be totally oblivious to where it's being used from.